src/backend/routers/iep.ts

```diff
@@ -92,6 +92,23 @@ export const iep = router({
     .mutation(async (req) => {
       const { subgoal_id, due_date, trial_count } = req.input;
       const { userId } = req.ctx.auth;
+
+      const shouldAdd = await req.ctx.db
+        .selectFrom("task")
+        .selectAll()
+        .where((eb) =>
+          eb.and([
+            eb("subgoal_id", "=", subgoal_id),
+            eb("assignee_id", "=", userId),
```

Comment:
should we have a database constraint on this to prevent duplicates?

Reply:
I think there might be a case where we allow duplicates in certain situations. Like if we want to redo the same benchmark or something. Let's hold off on that until we get the frontend designs and discuss that with the design team.